torchvision/prototype/transforms/functional/_geometry.py

```diff
@@ -40,12 +40,58 @@ def horizontal_flip_bounding_box(
     ).view(shape)
 
 
+def horizontal_flip(inpt: Any) -> Any:
```

Comment:
Why Any and not Union[Tensor, PIL.Image]?

Reply:
If we replace Any by `union[tensor, pil-image]` should we keep`else` branch ? In this case, it could be 
```
def horizontal_flip(inpt: Union[torch.Tensor, PIL.Image.Image]) -> Union[torch.Tensor, PIL.Image.Image]:
    if isinstance(inpt, features._Feature):
        return inpt.horizontal_flip()
    elif isinstance(inpt, PIL.Image.Image):
        return horizontal_flip_image_pil(inpt)
    # elif isinstance(inpt, torch.Tensor):
    #     return horizontal_flip_image_tensor(inpt)
    # else:
    #     return inpt
    else:
        return horizontal_flip_image_tensor(inpt)
```

Reply:
Agreed. As discussed earlier, the middle-layer shouldn't return the input as-is on else. I assume this will be removed on a follow up PR? Also we might want to play around with the order of the ifs and the type info, to see if there is any way we can make the middle layer JIT-scriptable.